dtfs-central-api/src/scheduler/jobs/create-utilisation-reports/index.ts

```diff
@@ -0,0 +1,60 @@
+import { asString } from '@ukef/dtfs2-common';
+import { getOneUtilisationReportDetailsByBankId, saveNotReceivedUtilisationReport } from '../../../services/repositories/utilisation-reports-repo';
+import { getAllBanks } from '../../../services/repositories/banks-repo';
+import { SchedulerJob } from '../../../types/scheduler-job';
+import { Bank } from '../../../types/db-models/banks';
+import { getCurrentReportPeriodForBankSchedule } from '../../../utils/report-period';
+
+const { UTILISATION_REPORT_CREATION_FOR_BANKS_SCHEDULE } = process.env;
+
+/**
+ * Checks if the current bank report is missing
+ * @param bank - The bank
+ * @returns Whether or not the bank report is missing
+ */
+const isCurrentBankReportMissing = async (bank: Bank): Promise<boolean> => {
+  const currentReportPeriod = getCurrentReportPeriodForBankSchedule(bank.utilisationReportPeriodSchedule);
+  const currentUtilisationReportForBank = await getOneUtilisationReportDetailsByBankId(bank.id, { reportPeriod: currentReportPeriod });
+  return !currentUtilisationReportForBank;
+};
+
+/**
+ * Gets the banks which are visible in TFM utilisation
+ * reports and do not have any report in the database
+ * @returns The banks which do not have reports
+ */
+const getBanksWithMissingReports = async (): Promise<Bank[]> => {
+  const banksVisibleInTfm = (await getAllBanks()).filter((bank) => bank.isVisibleInTfmUtilisationReports);
```

Comment:
A little worried we named this feature flag as just 'is visible in TFM' and now it's 'is opted into reporting', I think it's fine for now.

Reply:
Yeah, agreed. Make sense for this to be more general. Probably wouldn't be a big effort if we did want to change it?

Reply:
Yeha just requires a DB change in prod when it gets there which is as much effort as the change. Might be worth us doing anyway and I'll raise a separate ticket for that